packages/framework-integration-tests/integration/end-to-end/events.integration.ts

```diff
@@ -0,0 +1,405 @@
+                to: to.toISOString(),
+              },
+              mockCartId
+            )
+            const events: Array<EventSearchResponse> = result.data['eventsByEntity']
+            // First check the order and structure
+            checkOrderAndStructureOfEvents(events)
+            // Now we check that we have received more than 0 events and less than number we provisioned, as time filters
+            // we used should have given us less events than what we provisioned
+            expect(events.length).to.be.within(1, numberOfProvisionedEvents - 1)
+          })
+        })
+      })
+
+      context('when doing a query by type', () => {
+        context('with no time filters', () => {
+          it('returns the expected events in the right order', async () => {
+            const result = await queryByType(anonymousClient, 'CartItemChanged')
+            // As we are querying by just event type, we will get the events we provisioned plus others (possible MANY others)
+            // with different IDs. The only things we can check are:
+            // - That, at least, we have "numberOfProvisionedEvents" events
+            // - The structure is the expected one
+            // - They came sorted by "createdAt" in descenting order.
+            const events: Array<EventSearchResponse> = result.data['eventsByType'].slice(0, numberOfProvisionedEvents)
+            expect(events.length).to.be.equal(numberOfProvisionedEvents)
+            checkOrderAndStructureOfEvents(events)
+          })
+        })
+
+        context('with time filters', () => {
+          it('returns the expected events in the right order', async () => {
+            // The structure and the reasons of why this test is this way are exactly the same as described in tests:
+            // 'when doing a query by type'.'with time filters'.'returns the expected events in the right order'
+            const from = new Date(eventsProvisionedStartedAt)
+            from.setSeconds(from.getSeconds() - 1)
+            const to = new Date(eventsProvisionedFinishedAt)
+            to.setSeconds(to.getSeconds() + 1)
+
+            const result = await queryByType(anonymousClient, 'CartItemChanged', {
+              from: from.toISOString(),
+              to: to.toISOString(),
+            })
+            const events: Array<EventSearchResponse> = result.data['eventsByType']
+            // First check the order and structure
+            checkOrderAndStructureOfEvents(events)
+            // Now check if the time filter worked well
+            let provisionedEventsFound = 0
+            for (const event of events) {
+              if (event.entityID === mockCartId) {
+                provisionedEventsFound++
+              }
+            }
+            expect(provisionedEventsFound).to.be.gte(numberOfProvisionedEvents)
+          })
+        })
+      })
+    })
+    describe('the result of the queries involving many events', () => {
```

Comment:
In this test, we create 150 events concurrently. It was the one that surfaced the bug that was causing a loss of events. 

This motivated the change in the storage of events so that they now contain a UUID suffix in their sorting key